### PR TITLE
Inserting extra newline after pasting into a  terminal buffer

### PR DESCRIPTION
**Pull request by ghost**
Hello internet stranger,

I found an issue with using your project and white-space language REPLs in terminal buffers. Specifically, when you paste into a terminal buffer, you automatically add a '\r'  to the end of the selection. This works fine for most cases, but for white-space based languages (notably python) it assumes you're going to write another line rather than send the code to the REPL. I've changed this to add two newlines after a paste so the REPL knows to process the text rather than wait for more. This is the most language-agnostic way of solving the problem, but it has the small side effect of adding a newline after pasting into the terminal for languages that don't rely on white-space.

I think your project is great - it adds some cool IDE-like features without adding bloat. This is just I small bug I wanted to nip in the bud.

### CODE REVIEW

**Comment by karoliskoncevicius** [comment]
Hello,

Big thank you for the kind words and for looking into this plugin.

Before considering this change, however, I would like to see the issue at hand. Unfortunately I don't know first thing about python, however I tried it with a very simple script like this:

    a = 2
    b = 3

    a + b

And it seemed to work fine - after I sent `a = 2` it executed it and didn't wait for additional line.

For now I am a bit reluctant to add second `\r` to ALL terminals, as it would add a lot new empty lines. People typically have their terminals open on the bottom with very little window height (10 lines or so). So if every second line of that small window will become empty - it would make it hard to read what's there. However if needed I could include your pull req with some kind of check for the language type (so for example I am thinking about adding second `\r` if the line was sent from a `filtype=py` for example, which would solve the issue).

However, like mentioned - it would be nice if I could see the issue myself. Maybe you have a suggestion for the script I should look into in order to test this problem more?

**Comment by ghost** [comment]
Sorry about that, I should've gone more in depth. Here's an example of a script which doesn't work:
```
for i in range(30):
    print(i)
```
Here are some examples of me testing this script [without](https://asciinema.org/a/nggAxetr9o4hp9Vt4YtFCEhCK) and [with](https://asciinema.org/a/vlAjMMSGm3rzCZynKilEK2mTP) the patch.

As for using a filetype check, would it be smarter/more general-purpose to expose the ending characters to user input? It'd be trivial on the user's part to either bind a mapping or using an autocommand to change the plugin's setting on the fly.

**Comment by karoliskoncevicius** [comment]
Thank you a lot for the demonstration, I think I have a fuller understanding now.

So my problem with double `\r\r` in this case now is that it would prohibit users from sending a long `for` construct in chunks. Consider something like this:

    for i in range(x):
      doSomething()
      doSomethingElse()
      doOneMoreThing()
      doAnotherThing()
    
Now with this construct if you sent first 2 lines with `\r\r` - it would start executing, which is not intended behaviour. And if the script is written correctly - there will be an empty line at the end, which could also be sent to the REPL in order to start executing. So in cases like this:

    for i in range(30):
        print(i)
        <empty line>

There should be an empty line in the end, which could be sent to the REPL with (e.g) `<space>l$` (assuming default keymaps). Or even if one uses `<space>lap`, to send the whole paragraph, - the empty line after the paragraph will be sent to the REPL and it will be correctly executed.

Bottom line: Currently I am not in favour of adding `\r\r` for the reasons stated above. What do you think?

**Comment by ghost** [comment]
That makes sense. I've done some more testing, and I *think* the error stems from not testing for an end of line when pushing to a terminal. I've pushed a change to my PR for you to check out.

**Comment by karoliskoncevicius** [comment]
Hello again. After investigating this for a bit I honestly do not see a need for a change... I mean - when you have multiple lines indented in a script - there will be an empty line at the end. A person sending the script to terminal can always send the new line at the end in order to run the code.

Attaching a small example of how it is supposed to work. Note that adding '\r' would break the second example:

example

**Comment by ghost** [comment]
I see what you're getting at and I think you're right. I usually use the visual mapping with python (vip), but it makes sense that if you send it in line by line you'd want each line to get added before they're run. However, one edge case the current code misses is with endoflines in documents. If I have a file that looks like this:
```
<START OF FILE>
for i in range(30):
    print(i)
<END OF FILE>
```
That is perfectly valid python code, but it doesn't work on the current codebase. I'll close my PR for now and get back to you.

**Comment by karoliskoncevicius** [comment]
Just a quick comment - I think selecting with `vap` might work better in your case (instead of `vip`), as `vap` should also select new line after the paragraph.
